vmware-event-router/internal/processor/aws_event_bridge.go

```diff
@@ -171,102 +178,70 @@ func NewAWSEventBridgeProcessor(ctx context.Context, cfg connection.Config, sour
 	return &eventBridge, nil
 }
 
-// Process implements the stream processor interface TODO: handle
-// throttling/batching
-// https://docs.aws.amazon.com/eventbridge/latest/userguide/cloudwatch-limits-eventbridge.html#putevents-limits
-func (awsEventBridge *awsEventBridgeProcessor) Process(moref types.ManagedObjectReference, baseEvent []types.BaseEvent) error {
-	batchInput, err := awsEventBridge.createPutEventsInput(baseEvent)
-	if err != nil {
-		errMsg := fmt.Errorf("could not create PutEventsInput for event(s): %v", err)
-		awsEventBridge.Println(errMsg)
-		return processorError(ProviderAWS, errMsg)
-	}
-
-	// nothing to send
-	if len(batchInput) == 0 {
-		return nil
+// Process implements the stream processor interface
+func (awsEventBridge *awsEventBridgeProcessor) Process(ce cloudevents.Event) error {
+	if awsEventBridge.verbose {
+		awsEventBridge.Printf("processing event (ID %s): %v", ce.ID(), ce)
 	}
 
-	for idx, input := range batchInput {
-		resp, err := awsEventBridge.PutEvents(&input)
-		if err != nil {
-			awsEventBridge.Printf("could not send event(s) for batch %d: %v", idx, err)
-			continue
-		}
+	awsEventBridge.mu.RLock()
```

Comment:
Do we lock here since we process single cloudevent in Process now?

Reply:
Yes we do (and need the lock if that's the question) since there's potential concurrent access to the map due to the background goroutine syncing the rules.

Reply:
Yeah, ok I was just curious as this lock is new 👍

Reply:
Never mind the new, saw we use the lock in another method